cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/KafkaAnomaly.java

```diff
@@ -29,8 +33,35 @@ String optimizationResult(boolean isJson) {
     return isJson ? _optimizationResult.cachedJSONResponse() : _optimizationResult.cachedPlaintextResponse();
   }
 
+  /**
+   * Get the type of anomaly.
+   *
+   * @return The type of anomaly.
+   */
+  public abstract AnomalyType anomalyType();
+
+  /**
+   * Get the detection time of anomaly.
+   *
+   * @return The detection time of anomaly.
+   */
+  public long detectionTimeMs() {
```

Comment:
Can `detectionTimeMs` and `anomalyType` be moved to the `Anomaly` interface, as well? -- note that in this case, there should ideally be an interface for `AnomalyType`,as well. (please see `EndPoint` and `EndpointType` interfaces for similar use-cases).